### ISSUE TITLE
Requests to /zappa/socket/__local/... (fail)

### ISSUE THREAD

**Issue by rugk** [audreyt/ethercalc#350]
What is this request?

```
/zappa/socket/__local/zgFjzfQ9uhwqROQZAAAD
```

In my installation of Ethercalc in a subdir this file is always requested from my 
domain root. So it should actually fetch them from `CustomSubDir/zappa/...`, but it tries to fetch them from `/zappa/...`.
BTW: Is this also a websocket?

I'm using `--basedir` and also some nginx rewrites (`proxy_redirect`).

**Comment by audreyt** [comment]
Yes it is a websocket. Try using `--polling` to disable that message.

**Comment by rugk** [comment]
No websockets are okay. The issue is just that it is not rewritten correctly. As said I use `--basedir`, but @ZappaJS still uses the wrong URL for fetching the 

So please reopen this issue. It may also be an issue on @ZappaJS's side. So am I right that the `basedir` option should add the specified dir to the request path?

**Comment by rugk** [comment]
So this is a workaround for nginx:

```nginx
# workaround for https://github.com/audreyt/ethercalc/issues/350
location /zappa/socket/__local/ {
        rewrite (.*) /CustomSubDir/$1;
}
```

**Comment by rugk** [comment]
It's still a workaround. The underlying issue should be fixed.
So I advocate for opening this issue again.

**Comment by audreyt** [comment]
Sure! Feel free to work on it. :-)

**Comment by rugk** [comment]
I have no idea what to change, but maybe someone else knows why @ZappaJS does not use the correct URL...

**Comment by sbraz** [comment]
Hi @audreyt, is there any chance you could look at this old issue which is still relevant today?

I have been changing `zappa_prefix` in `zappajs/lib/zappa.js` but it's a terrible workaround which also requires more URL rewriting on the reverse proxy.

I also noticed that calls to these `/calc/zappa/socket/…` URLs always return the following error, even with the demo https://ethercalc.net/ site:
```json
{
  "error": "No session"
}
```

Doesn't it mean that they could simply be dropped?

**Comment by audreyt** [comment]
That's likely, as we do not make use of cookie-based sessions.

A pull request to remove the call altogether would be appreciated.

**Comment by sbraz** [comment]
Thanks for the feedback. I have looked at the code but I don't have any experience with server-side JS and I haven't found a way to disable this.

**Comment by audreyt** [comment]
The TypeScript rewrite drops zappajs in favor of Hono on Cloudflare Workers ([CLAUDE.md §3.1](https://github.com/audreyt/ethercalc/blob/sandstorm/CLAUDE.md)). zappajs upgrades, Node-version compatibility, and `/zappa/socket/*` routes no longer apply. Closing.
